api/v1alpha4/clusterclass_types.go

```diff
@@ -53,6 +53,23 @@ type ClusterClassSpec struct {
 	Workers WorkersClass `json:"workers,omitempty"`
 }
 
+// ControlPlaneClass defines the class for the control plane.
+type ControlPlaneClass struct {
+	Metadata ObjectMeta `json:"metadata,omitempty"`
+
+	// LocalObjectTemplate contains the reference to the control plane provider.
+	LocalObjectTemplate `json:",inline"`
```

Comment:
I am probably missing something but the `LocalObjectTemplate` we have here refs to a ControlPlaneTemplate (Ex: KubeadmControlPlaneTemplate - will be introduced in https://github.com/kubernetes-sigs/cluster-api/pull/4904 ).

This ControlPlaneTemplate will internally refer to a machine infrastructure, isnt it? 
Going back the kcp template example we have a machine infra ref at `kcptemplate.spec.template.spec.machineTemplate.infrastructureRef`.   How is this different from the new `MachineInfrastructure` field?

Reply:
Imho both will reference the same InfrastructureMachineTemplate. This PR makes it explicit on the ClusterClass (i.e. you have to set the ref on the ClusterClass **and** inside the ControlPlaneTemplate)

Reply:
Exactly, we are making this explicit;
Also, according to the initial work on https://github.com/kubernetes-sigs/cluster-api/pull/5002, ClusterClass is always going to set the reference inside the ControlPlaneTemplate, so it is not required to set it in both places upfront

Reply:
Good that fits for my PR as I assume that the ref can be unset in the ControlPlane even if it is set in the ClusterClass.

Reply:
>ClusterClass is always going to set the reference inside the ControlPlaneTemplate, so it is not required to set it in both places upfront

Good to know that we are doing this, however the `kcptemplate.spec.template.spec.machineTemplate.infrastructureRef` is not an optional field. So are we expecting that the user set the filed **both** inside the ControlPlaneTemplate and inside the ClusterClass definition and also that both the fields point to the same thing?
We should throw an error incase they dont match, is it?